src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -166,6 +173,7 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
   private object EmptyHashMap extends HashMap[Any, Nothing] { 
     override def head: (Any, Nothing) = throw new NoSuchElementException("Empty Map")
     override def tail: HashMap[Any, Nothing] = throw new NoSuchElementException("Empty Map")
+    override val hashCode: Int = super.hashCode()
```

Comment:
This val changes the automatically generated serial version UID.

```
➜  scala-2.12.x git:(topic/tsan-cleanup) ✗ scala-launch 2.12.9 -nc -e 'println(java.io.ObjectStreamClass.lookup(scala.collection.immutable.HashMap.empty[Any, Any].getClass))'
scala.collection.immutable.HashMap$EmptyHashMap$: static final long serialVersionUID = 2121866088229452155L;
➜  scala-2.12.x git:(topic/tsan-cleanup) ✗ scala-pr 8399 -nc -e 'println(java.io.ObjectStreamClass.lookup(scala.collection.immutable.HashMap.empty[Any, Any].getClass))'
scala.collection.immutable.HashMap$EmptyHashMap$: static final long serialVersionUID = -2471795291674749849L;
```

In 2.13.x we consistently use the serialization proxy pattern to avoid this instability. For 2.12.x, you could either revert this line, or replace it with a def returning a hard-coded constant that is proven to be the right value by a unit test.

Reply:
took the third option - a def and a test